File: be-engine-v1/be-security/src/main/java/com/cmcorg20230301/be/engine/security/model/enums/SysRequestCategoryEnum.java
package com.cmcorg20230301.be.engine.security.model.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@AllArgsConstructor
@Getter
@Schema(description = "请求类别")
public enum SysRequestCategoryEnum {

    PC_BROWSER_WINDOWS(101, "windows-浏览器"), //
    PC_BROWSER_MAC(102, "mac-浏览器"), //
    PC_BROWSER_LINUX(103, "linux-浏览器"), //

    PC_BROWSER_WINDOWS_WX(104, "windows-浏览器-微信"), //
    PC_BROWSER_MAC_WX(105, "mac-浏览器-微信"), //
    PC_BROWSER_LINUX_WX(106, "linux-浏览器-微信"), //

    PC_CLIENT_WINDOWS(201, "windows-客户端"), //
    PC_CLIENT_MAC(202, "mac-客户端"), //
    PC_CLIENT_LINUX(203, "linux-客户端"), //

    ANDROID(301, "安卓端"), //
    ANDROID_BROWSER(302, "安卓-浏览器"), //
    ANDROID_BROWSER_WX(303, "安卓-浏览器-微信"), //

    IOS(401, "苹果端"), //
    IOS_BROWSER(402, "苹果-浏览器"), //
    IOS_BROWSER_WX(403, "苹果-浏览器-微信"), //

    MINI_PROGRAM_WE_CHAT_ANDROID(501, "小程序-微信-安卓"), //
    MINI_PROGRAM_WE_CHAT_IOS(502, "小程序-微信-苹果"), //

    OFFICIAL_ACCOUNT_WX(601, "公众号-微信"), //

    ;

    @EnumValue
    @JsonValue
    private final int code;
    private final String name;

    @NotNull
    public static SysRequestCategoryEnum getByCode(@Nullable Integer code) {

        if (code == null) {
            return PC_BROWSER_WINDOWS;
        }

        for (SysRequestCategoryEnum item : SysRequestCategoryEnum.values()) {

            if (item.getCode() == code) {

                return item;

            }

        }

        return PC_BROWSER_WINDOWS;

    }

}

File: fe-antd-v1/src/model/enum/SysRequestCategoryEnum.ts
import {IEnum} from "@/model/enum/CommonEnum";
import {ProSchemaValueEnumType} from "@ant-design/pro-components";

export interface ISysRequestCategoryEnum {

    PC_BROWSER_WINDOWS: IEnum<number>,
    PC_BROWSER_MAC: IEnum<number>,
    PC_BROWSER_LINUX: IEnum<number>,

    PC_BROWSER_WINDOWS_WX: IEnum<number>,
    PC_BROWSER_MAC_WX: IEnum<number>,
    PC_BROWSER_LINUX_WX: IEnum<number>,

    PC_CLIENT_WINDOWS: IEnum<number>,
    PC_CLIENT_MAC: IEnum<number>,
    PC_CLIENT_LINUX: IEnum<number>,

    ANDROID: IEnum<number>,
    ANDROID_BROWSER: IEnum<number>,
    ANDROID_BROWSER_WX: IEnum<number>,

    IOS: IEnum<number>,
    IOS_BROWSER: IEnum<number>,
    IOS_BROWSER_WX: IEnum<number>,

    MINI_PROGRAM_WE_CHAT_ANDROID: IEnum<number>,
    MINI_PROGRAM_WE_CHAT_IOS: IEnum<number>,

    OFFICIAL_ACCOUNT_WX: IEnum<number>,

}

// 请求类别，枚举类
export const SysRequestCategoryEnum: ISysRequestCategoryEnum = {

    PC_BROWSER_WINDOWS: {
        code: 101,
        name: 'windows-浏览器',
        browserFlag: true,
    },
    PC_BROWSER_MAC: {
        code: 102,
        name: 'mac-浏览器',
        browserFlag: true,
    },
    PC_BROWSER_LINUX: {
        code: 103,
        name: 'linux-浏览器',
        browserFlag: true,
    },

    PC_BROWSER_WINDOWS_WX: {
        code: 104,
        name: 'windows-浏览器-微信',
        browserFlag: true,
    },
    PC_BROWSER_MAC_WX: {
        code: 105,
        name: 'mac-浏览器-微信',
        browserFlag: true,
    },
    PC_BROWSER_LINUX_WX: {
        code: 106,
        name: 'linux-浏览器-微信',
        browserFlag: true,
    },

    PC_CLIENT_WINDOWS: {
        code: 201,
        name: 'windows-客户端',
    },
    PC_CLIENT_MAC: {
        code: 202,
        name: 'mac-客户端',
    },
    PC_CLIENT_LINUX: {
        code: 203,
        name: 'linux-客户端',
    },

    ANDROID: {
        code: 301,
        name: '安卓端',
    },
    ANDROID_BROWSER: {
        code: 302,
        name: '安卓-浏览器',
        browserFlag: true,
    },
    ANDROID_BROWSER_WX: {
        code: 303,
        name: '安卓-浏览器-微信',
        browserFlag: true,
    },

    IOS: {
        code: 401,
        name: '苹果端',
    },
    IOS_BROWSER: {
        code: 402,
        name: '苹果-浏览器',
        browserFlag: true,
    },
    IOS_BROWSER_WX: {
        code: 403,
        name: '苹果-浏览器-微信',
        browserFlag: true,
    },

    MINI_PROGRAM_WE_CHAT_ANDROID: {
        code: 501,
        name: '小程序-微信-安卓',
    },
    MINI_PROGRAM_WE_CHAT_IOS: {
        code: 502,
        name: '小程序-微信-苹果',
    },

    OFFICIAL_ACCOUNT_WX: {
        code: 601,
        name: '公众号-微信',
    },

}

export const SysRequestCategoryEnumDict = new Map<number, ProSchemaValueEnumType>();

Object.keys(SysRequestCategoryEnum).forEach(key => {

    const item = SysRequestCategoryEnum[key];

    SysRequestCategoryEnumDict.set(item.code as number, {text: item.name})

})
